lib/Server.js

```diff
@@ -588,14 +588,18 @@ class Server {
     const useColor = getColorsOption(getCompilerConfigArray(this.compiler));
 
     const protocol = this.options.https ? 'https' : 'http';
-    const { hostname, port } = this;
+    const { address: hostname, port } = this.server.address();
     const prettyPrintUrl = (newHostname) =>
       url.format({ protocol, hostname: newHostname, port, pathname: '/' });
 
     let prettyHostname;
     let lanUrlForTerminal;
 
-    if (hostname === '0.0.0.0' || hostname === '::') {
+    if (
+      hostname === '0.0.0.0' ||
+      hostname === '127.0.0.1' ||
```

Comment:
In theory `127.0.0.1` can be not localhost (very rare edge case), also localhost can be not `127.0.0.1` (also very rare edge case). Why we added this?

Reply:
because without it we use `webpack serve --host localhost` it was printing `server running at http://127.0.0.1:8080` instead of `http://localhost:8080`. also failing some snapshots. I thought they should be the same.

![Screenshot at 2021-02-15 19-03-26](https://user-images.githubusercontent.com/46647141/107953110-a5bd9080-6fc0-11eb-8384-009a8b190275.png)

Reply:
Here right solution https://github.com/indutny/node-ip/blob/master/lib/ip.js#L320 just use `ip.isLoopback()`

Reply:
Should be easy to fix :star:

Reply:
Like this ?

```suggestion
      ip.isLoopback(hostname) ||
```

Reply:
Yep, we also can remove `hostname === '::'` and maybe `0.0.0.0` need test it (manually as minimum)